cmd/kubeadm/app/phases/upgrade/compute.go

```diff
@@ -112,13 +112,17 @@ func GetAvailableUpgrades(versionGetterImpl VersionGetter, experimentalUpgradesA
 		return nil, err
 	}
 
+	if !etcdStatus.VersionsMatch() {
```

Comment:
I don't think this belongs here. This function is just about getting a current state of the world.  Maybe it could go in enforceRequirements?

Reply:
I agree that this is not ideal. This is merely addressing how we contact external etcd nodes.

We have preflight checks that address etcd cluster versions, and they do not use the code that I am introducing here. In fact, they use an HTTP client to probe an ectd cluster.  I believe that we should fix those preflight checks to make use of this etcd interrogator code, but that would be expanding the scope of this PR significantly.  I am happy to open a follow-up issue here, but in this PR trying to make the change a bit more manageable.  Thoughts?

Reply:
I'm ok with the failing here, b/c versions on inconsistent.